pkg/config/app_configuration.go

```diff
@@ -13,6 +13,12 @@ type ApplicationConfig struct {
 	// Duration. example: "30s"
 	ActorScanInterval string `json:"actorScanInterval"`
 	// Duration. example: "30s"
-	DrainOngoingCallTimeout string `json:"drainOngoingCallTimeout"`
-	DrainRebalancedActors   bool   `json:"drainRebalancedActors"`
+	DrainOngoingCallTimeout string           `json:"drainOngoingCallTimeout"`
+	DrainRebalancedActors   bool             `json:"drainRebalancedActors"`
+	Reentrancy              ReentrancyConfig `json:"reentrancy,omitempty"`
+}
+
+type ReentrancyConfig struct {
+	Enabled       bool `json:"enabled"`
+	MaxStackDepth *int `json:"maxStackDepth,omitempty"`
```

Comment:
@halspang Is there a reason why the MaxStackDepth needs to be a an int pointer?

Reply:
It's to allow it to be actually empty instead of taking the int default. Even though the default is 0, we can't tell when we parse the config if that was the intent of the author or not since the json serializer set it for us.

Granted, setting it to 0 would effectively disable your actor, so there's no reason to do so but I felt like this was more defensive.

Reply:
@mukundansundar This is pretty common strategy in Crossplane source code, we don't use it as much. I am fine with this.